main.md

```diff
@@ -187,20 +184,20 @@ This is illustrated by the following example
 
 The AS MUST process the request as follows:
 
-1. The AS MUST authenticate the client in same way as at the token endpoint.
-2. The AS MUST must reject the request if the `request_uri` authorization request parameter is provided.
-3. The AS MUST validate the request the same way as at the authorization endpoint. For example, the authorization server checks whether the redirect URI matches one of the redirect URIs configured for the `client_id`. It MAY also check whether the client is authorized for the `scope` for which it is requesting access. This validation allows the authorization server to refuse unauthorized or fraudulent requests early. 
+1. The AS MUST authenticate the client in the same way as at the token endpoint.
+2. The AS MUST reject the request if the `request_uri` authorization request parameter is provided.
+3. The AS MUST validate the request in the same way as at the authorization endpoint. For example, the authorization server checks whether the redirect URI matches one of the redirect URIs configured for the client. It MAY also check whether the client is authorized for the `scope` for which it is requesting access. This validation allows the authorization server to refuse unauthorized or fraudulent requests early.
```

Comment:
Suppose

- Your social security number is "15 070649 C 103".
- You have cars.

Which sentence below do you think sounds more natural?

- The cars belong to the social security number.
- The cars belong to you.

Conceptually,

- You = Client
- Social Security Number = client_id
- Cars = Request URIs

Both request URIs and `client_id` are attributes of a client. _"One attribute is bound to another attribute"_ sounds a bit weird.

Client IDs are not always primary keys in the client DB table in AS implementations. In addition, some implementations allow clients to have multiple client identifiers.

Reply:
I agree that `client` is the object that everything is associated with, whereas `client_id` is a specific public identifier (perhaps one of several) that identifies the client. Therefore, I think it reads better to have `client` in the text unless you're talking about comparing the `client_id` string precisely.

Reply:
How is the client supposed to know what client ids belong to the same conceptual client? I'm asking because this governs the redirect URIs it is allowed to use here.

Your arguments are all reasonable but I want to make sure we write a spec that fosters interop.

Reply:
I find 'client' is clearer here; that's the language used in 6749 in general, though some of the other specs (e.g. OpenID Connect) do seem to occasionally use client_id in this way.

With an AS vendor hat on, the current text does imply/outright state that I should explicitly bind the returned request_uri to exactly the client_id that was used to register it, rather than to the overall "client" - which (given the AS I'm currently thinking of technically allows the client_id to be changed via the management interface available to the AS owner) would require at least an extra db column or table to be technically compliant with this clause, and it's not clear that this would provide any gain. (Although it does also feel like that's a bit of an esoteric situation that in reality I probably shouldn't worry about...)

Reply:
There exists at least one AS implementation that allows a client to have a client ID and an alias of it. This feature is useful when a customer replaces their current AS implementation with a new one but wants to continue to use client IDs that have been assigned by the previous AS implementation.

AS implementations that support such feature will have a `client_id` column and a `alias` column in the definition of the client DB table. It depends on its implementation, but the implementation can allow clients to make authorization requests with either `client_id` or `alias`.

Some implementations may not enable clients to have an alias, but still they may be able to allow clients to change client IDs.

If an AS implementation (1) has 'alias' feature and/or (2) allows clients to change client IDs, the requirement "request URI should be bound to `client_id`" would impose some restrictions on its implementation, but such implementation details should be left to implementers.

Reply:
I see. Thanks for the explanation. I'm fine with the suggested text since it ensure interop as long as the client uses the same client_id but leaves the door open for vendor/deployment-specific features.